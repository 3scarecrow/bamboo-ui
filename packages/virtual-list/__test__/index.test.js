import path from 'path';
import simulate from 'miniprogram-simulate';

test('should render demo and match snapshot', () => {
  const id = simulate.load(path.resolve(__dirname, '../demo/index'), {
    rootPath: path.resolve(__dirname, '../../'),
  });
  const comp = simulate.render(id);
  comp.attach(document.createElement('parent-wrapper'));
  expect(comp.toJSON()).toMatchSnapshot();
});


// // import path from 'path';
// // import simulate from 'miniprogram-simulate';

// // test('should render demo and match snapshot', () => {
// //   const id = simulate.load(path.resolve(__dirname, '../index'), {
// //     rootPath: path.resolve(__dirname, '../../'),
// //   });
// //   const comp = simulate.render(id);
// //   comp.attach(document.createElement('parent-wrapper'));
// //   expect(comp.toJSON()).toMatchSnapshot();
// // });

// import path from 'path'
// import simulate from 'miniprogram-simulate'

// describe('actionsheet', () => {
//     let id

//     beforeAll(() => {
//         id = simulate.load(path.resolve(__dirname, './index'), {
//                 rootPath: path.resolve(__dirname, '../../'),
//             }
//         )
//     })

//     test('basic actionsheet',() => {
//         const container = simulate.render(id)
//         container.attach(document.createElement('parent-wrapper'))

//         console.log('container.toJSON()', container.toJSON())
//         expect(container.toJSON()).toMatchSnapshot()
//     })

//     // test('show on create', async () => {
//     //     const container = simulate.render(id, { show: true })
//     //     container.attach(document.createElement('parent-wrapper'))

//     //     expect(container.toJSON()).toMatchSnapshot()
//     // })

//     // test('no cancel', async () => {
//     //     const container = simulate.render(id, { showCancel: false })
//     //     container.attach(document.createElement('parent-wrapper'))

//     //     expect(container.toJSON()).toMatchSnapshot()
//     // })

//     // test('touch', async () => {
//     //     const container = simulate.render(id)
//     //     container.attach(document.createElement('parent-wrapper'))
//     //     const comp = container.querySelector('.comp')

//     //     const menu = comp.querySelectorAll('.weui-actionsheet__menu .weui-actionsheet__cell')

//     //     menu[2].dispatchEvent('touchstart')
//     //     menu[2].dispatchEvent('touchend')
//     //     await simulate.sleep(10)
//     //     expect(container.data.tapValue).toBe(3)

//     //     menu[1].dispatchEvent('touchstart')
//     //     menu[1].dispatchEvent('touchend')
//     //     await simulate.sleep(10)
//     //     expect(container.data.tapValue).toBe(2)
//     // })

//     // test('close', async () => {
//     //     const container = simulate.render(id)
//     //     container.attach(document.createElement('parent-wrapper'))
//     //     const comp = container.querySelector('.comp')

//     //     const cancel = comp.querySelector('.weui-actionsheet__action .weui-actionsheet__cell')

//     //     cancel.dispatchEvent('touchstart')
//     //     cancel.dispatchEvent('touchend')
//     //     await simulate.sleep(10)
//     //     expect(container.data.show).toBe(false)
//     //     expect(comp.data.show).toBe(false)
//     // })
// })