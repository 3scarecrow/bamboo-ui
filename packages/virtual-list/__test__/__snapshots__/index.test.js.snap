// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render demo and match snapshot 1`] = `
<main>
  <virtual-list
    endIndex=""
    height="1000rpx"
    id="virtual-list"
    itemHeight="250rpx"
    refresherDefaultStyle="none"
    refresherEnabled="{{true}}"
    refresherTriggered="{{false}}"
    sourceData="{{Array []}}"
    startIndex=""
    bind:load="load"
    bind:refresh="refresh"
    bind:visibleDataChange="onVisibleDataChange"
  >
    <wx-view
      slot="refresher"
    >
      刷新
    </wx-view>
    <wx-view
      slot="footer"
      style="text-align: center;"
    />
    <wx-view
      slot="empty"
      style="text-align: center;"
    >
      
    暂无数据
  
    </wx-view>
  </virtual-list>
</main>
`;
